CONTRIBUTING.md

```diff
@@ -0,0 +1,333 @@
+
+```plaintext
+    ```[javascript]
+    const channel = realtime.channels.get('channelName');
+    ```
+    ```[objc]
+    ARTRealtimeChannel *channel = [realtime.channels get:@"channelName"];
+    ```
+```
+
+To use nested codeblocks when describing features that are available to the realtime and REST interfaces, use the the same syntax as a normal codeblock but prefix the languages with `realtime_` or `rest_`. Spacing should be continuous as with a normal codeblock, however the snippets for each interface should all be written together, rather than interspersed:
+
+```plaintext
+    ```[realtime_javascript]
+    const channel = realtime.channels.get('channelName');
+    ```
+    ```[realtime_objc]
+    ARTRealtimeChannel *channel = [realtime.channels get:@"channelName"];
+    ```
+
+    ```[rest_javascript]
+    const channel = rest.channels.get('channelName');
+    ```
+    ```[rest_objc]
+    ARTRestChannel *channel = [realtime.channels get:@"channelName"];
+    ```
+```
+
+### In-line code
+
+In-line code should be written between `@` symbols. For example, `the @get()@ method`.
+
+### Variables
+
+The following variables can be inserted into codeblocks to generate values at runtime:
+
+| Variable | Description |
+| -------- | ----------- |
+| `{{API_KEY}}` | Injects a demo API key into the code sample if a user isn't logged in. Enables users to select an API key from from their dashboard if they are logged in. |
+| `{{RANDOM_CHANNEL_NAME}}` | Generates a random channel name. |
+
+### Definition lists
+
+Definition lists are used to display a list of terms and their descriptions, such as the available capabilities.
+
+They use the following syntax:
+
+```plaintext
+- publish := can publish messages to channels
+- presence := can register presence on a channel (enter, update and leave)
+```
+
+### Tables
+
+Textile tables are supported which are slightly more flexible than Markdown ones. They don't need to use a row of hyphens to separate the header and content either.
+
+For the header row use the following syntax:
+
+```plaintext
+|_. Flag |_. Description |
+| SUBSCRIBE | Can subscribe to receive messages on the channel. |
+```
+
+To make a cell span two columns:
+
+```plaintext
+|_/2=. Limit |
+|_. Soft |_. Hard |
+```
+
+### Admonitions
+
+There are three types of admonition that can be used; `note`, `important` and `further-reading`. Update the value of `data-type` to switch between them. Admonitions are written using the HTML `<aside>` tag. Content must be constructed in HTML, other than links and in-line code styling which accept textile format.
+
+```html
+<aside data-type='note'>
+<p>The following restrictions apply to "channel":/channels names and namespaces:</p>
+<ul><li>They are case sensitive</li>
+<li>They can't start with @[@ or @:@</li>
+<li>They can't be empty</li>
+<li>They can't contain newline characters or wildcards</li></ul>
+</aside>
```

Comment:
Maybe add.... "Ensure <aside> content is kept concise (to avoid long paragraphs inside notes)." ??